### ISSUE TITLE
Add more Lua libraries

### ISSUE THREAD

**Issue by SolarcGit** [adriengivry/Overload#246]
We should consider adding the following libraries to the `ScriptInterpreter.cpp`:

- `sol::lib::package`
- `sol::lib::coroutine`
- `sol::lib::string`
- `sol::lib::os`
- `sol::lib::table`

**Comment by adriengivry** [comment]
Good point. We should also evaluate the consequences of adding these libraries:
- Runtime overhead?
- Loading times?

**Comment by adriengivry** [comment]
> What these libraries add to the making of a game outweights the consequences In my opinion. (I am pretty new though please keep that in mind) Having these libraries, especially table, coroutine, string and os, would greatly help in developing a game.

That's for sure! But it's important to gather factual data about the consequences of making such a change in order to make a more informed decision.

**Comment by SolarcGit** [comment]
Yes opinion is not all. Sadly I dont know how to gather the data to find out if it would be detrimental to the performance. Is there any way I can push this?

**Comment by SolarcGit** [comment]
Is there any way I can help do that?

**Comment by adriengivry** [comment]
> Is there any way I can help do that?

For sure! You can do this investigation process, do the change first (include the libs), and measure the performance impact (compilation and runtime), and document all that 👌

**Comment by SolarcGit** [comment]
using sol2?

**Comment by adriengivry** [comment]
Overload is using sol2 for lua scripting yes

**Comment by SolarcGit** [comment]
ok ok got it but where do I start?

**Comment by adriengivry** [comment]
> ok ok got it but where do I start?

Make the requested change in `ScriptInterpreter.cpp`, and get some profiling data with VS. Dig into sol2 and investigate the implications of such a change.

**Comment by SolarcGit** [comment]
can you show me how I can do the change?

**Comment by adriengivry** [comment]
https://github.com/adriengivry/Overload/blob/a56596b45935660cd5d8e0e61f52f04adf756931/Sources/Overload/OvCore/src/OvCore/Scripting/ScriptInterpreter.cpp#L32

Read the documentation of the `open_libraries` function in sol2.

**Comment by SolarcGit** [comment]
will do!
